openapi-cli/src/test/java/io/ballerina/openapi/cmd/OpenAPICmdTest.java

```diff
@@ -80,68 +80,188 @@ public void testSuccessfulServiceGeneration() throws IOException {
         new CommandLine(cmd).parseArgs(args);
         cmd.execute();
         Path expectedSchemaFile = resourceDir.resolve(Paths.get("expected_gen", "petstore_schema.bal"));
-
         Stream<String> expectedSchemaLines = Files.lines(expectedSchemaFile);
         String expectedSchemaContent = expectedSchemaLines.collect(Collectors.joining("\n"));
         expectedSchemaLines.close();
         if (Files.exists(this.tmpDir.resolve("client.bal")) &&
                 Files.exists(this.tmpDir.resolve("petstore_service.bal")) &&
-                Files.exists(this.tmpDir.resolve("types.bal"))) {
+                Files.exists(this.tmpDir.resolve("types.bal")) &&
+                Files.exists(this.tmpDir.resolve("tests/test.bal"))) {
             //Compare schema contents
             Stream<String> schemaLines = Files.lines(this.tmpDir.resolve("types.bal"));
             String generatedSchema = schemaLines.collect(Collectors.joining("\n"));
             schemaLines.close();
+            generatedSchema = (generatedSchema.trim()).replaceAll("\\s+", "");
+            expectedSchemaContent = (expectedSchemaContent.trim()).replaceAll("\\s+", "");
+            if (expectedSchemaContent.equals(generatedSchema)) {
+                Assert.assertTrue(true);
+                deleteGeneratedFiles(false);
+            } else {
+                Assert.fail("Expected content and actual generated content is mismatched for: "
+                        + petstoreYaml.toString());
+                deleteGeneratedFiles(false);
+            }
+        } else {
+            Assert.fail("Service generation failed.");
```

Comment:
Shall we say failure reason as well? There are multiple conditions.

Reply:
In the implementation exceptions are immediately thrown at each failure with the stack trace. If code was not generated without throwing any exception, the reason for not generating is not clear. That is why we have added a generalized message here. @lnash94 is there a chance specify the error message here ?

Reply:
yeah, there is a chance to access command execute output by calling the `readOutput` function. Could you Please check this?  
```suggestion
Assert.fail("Service generation failed. " + readOutput(true));
```

Reply:
Added more detailed message in each test scenario.